compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2614,7 +2614,6 @@ public void visit(BLangWildCardMatchPattern wildCardMatchPattern) {
         }
         BType matchExprType = wildCardMatchPattern.matchExpr.type;
         if (types.isAssignable(matchExprType, symTable.anyType)) {
-            wildCardMatchPattern.matchesAll = true;
```

Comment:
Since we removed `matchesAll` and rely on `isLastPattern` now, shouldn't we set `isLastPattern` to true here?

Reply:
It is set in CodeAnalyser because it is related to code analysing.